### ISSUE TITLE
[d3d9] Worms Armageddon:  main menu keeps shaking if window mode is disabled

### ISSUE THREAD

**Issue by ranplayer** [doitsujin/dxvk#4104]
It seems the developers of Worms Armageddon implemented a minor workaround to boot the game in a small window mode when running it through Wine to prevent the issue described on the title.

![wa_boot_popup](https://github.com/doitsujin/dxvk/assets/69990640/0ff98545-3ed2-4691-a507-ec7b31c39b65)

If you select "No" or changed it later via the Graphics settings, then the main menu keeps shaking and you can't interact with it. The game's default renderer is D3D9. If you use Wine3D3 (`PROTON_USE_WINED3D=1 %command%`) the problem doesn't happen (or if you change the API to OpenGL (shader) via the Graphics settings)

### Software information
Game: Worms Armageddon
SteamId: 217200
Settings: default

### System information
- GPU: RX 6700 XT
- Driver: Mesa 24.1.3
- Wine version:  Proton 9.0-2
- DXVK version:  Proton's build

### Apitrace file(s)
- [API trace attached](https://mega.nz/file/wkMBEbiZ#OEzt4MMMi3ndKVELveN5Y7k2DMB4Ipj5kgUyBANc11k)

### Log files
- Proton logs: 
[steam-217200.tar.gz](https://github.com/user-attachments/files/16117050/steam-217200.tar.gz)

**Comment by K0bin** [comment]
There's nothing shaking when replaying the apitrace. Did you encounter the issue when you made the apitrace?

**Comment by ranplayer** [comment]
yes... I've retested here and found out the "shaking" just happens on Wayland (`xwayland 23.2.6`).  I've just noticed two issues for D3D9 renderer on X11:

1) the main menu is misaligned (check pictures below)

DXVK :
![wa_x11](https://github.com/doitsujin/dxvk/assets/69990640/74d9eec1-8687-453d-8262-b088b6548f42)


WineD3D:
![wa_wd3d](https://github.com/doitsujin/dxvk/assets/69990640/ec5782f2-d6bb-4141-b2dc-ce4e6e079a62)

2) If you alt+tab and focus the window later it turns completely black (attaching a trace: https://mega.nz/file/AotDTZLL#aIW3aY0TslXvL9rNW4egTDDs0DbvnsesW80p0zee5Ck).


(I'll just stopping using Wayland for testing games and wait for the proper Wine support. Sorry...)

**Comment by Blisto91** [comment]
Does launching with `DXVK_CONFIG="d3d9.deviceLossOnFocusLoss = True"` help with the alt-tab black screen

**Comment by ranplayer** [comment]
@Blisto91 it doesn't.  For WineD3D it just crashes instead of displaying the black screen

**Comment by ranplayer** [comment]
I'm reattaching the logs for the black screen on X11: 
[steam-217200-x11.tar.gz](https://github.com/user-attachments/files/16146071/steam-217200-x11.tar.gz)

It has several repeated logs of:
```
info:  Presenter: Actual swap chain properties:
info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
info:    Color space:  VK_COLOR_SPACE_SRGB_NONLINEAR_KHR
info:    Present mode: VK_PRESENT_MODE_FIFO_KHR (dynamic: yes)
info:    Buffer size:  160x31
info:    Image count:  4
info:    Exclusive FS: 0
```

**Comment by K0bin** [comment]
This really sounds like either a compositor bug or an XWayland bug. Which compositor/desktop environment are you using?

**Comment by ranplayer** [comment]
Kwin (Plasma 6.1)

**Comment by AlexFolland** [comment]
Can you please attach a video of this happening?  It would be good to see the exact behavior visually to understand it more clearly.

**Comment by ranplayer** [comment]
@AlexFolland sure... this is a video from the wayland session: 
[worms_armageedon_wayland.tar.gz](https://github.com/user-attachments/files/16181858/worms_armageedon_wayland.tar.gz)

**Comment by ranplayer** [comment]
folks, the "shaking issue" on wayland has been solved with the recent xwayland update (`24.1.1`)
